### PR TITLE
JAMES-2272 split WorkQueue as a standalone component

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by mbaechler** [comment]
test this please

**Comment by Arsnael** [comment]
```
RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:365 » ConditionTimeout
```
test this please

File: server/task/src/main/java/org/apache/james/task/WorkQueue.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+import com.google.common.collect.Sets;
+import reactor.core.Disposable;
+import reactor.core.publisher.WorkQueueProcessor;
+import reactor.core.scheduler.Schedulers;
+
+public class WorkQueue implements Closeable {
```
Comment (chibenwa) [inline server/task/src/main/java/org/apache/james/task/WorkQueue.java]: Isn't this more an InVmWorkQueue ?

Shouldn't we have an API for this class?

Comment (mbaechler) [reply server/task/src/main/java/org/apache/james/task/WorkQueue.java]: No. I only have one implementation for the moment, I don't want to extract an interface if it's not required.
I will be required in a later task but we should always implement as little as possible.

Comment (chibenwa) [reply server/task/src/main/java/org/apache/james/task/WorkQueue.java]: Sounds good to me

File: server/task/src/main/java/org/apache/james/task/MemoryTaskManagerWorker.java [hunk 2]
```diff
@@ -35,6 +33,8 @@
 import reactor.core.publisher.Mono;
 
 public class MemoryTaskManagerWorker implements TaskManagerWorker {
+
+
```
Comment (blackheaven) [inline server/task/src/main/java/org/apache/james/task/MemoryTaskManagerWorker.java]: Should not it be only one?

Comment (mbaechler) [reply server/task/src/main/java/org/apache/james/task/MemoryTaskManagerWorker.java]: I can change that but will not wait for the build to be green if you agree

File: server/task/pom.xml [hunk 3]
```diff
@@ -55,6 +55,10 @@
             <artifactId>junit-vintage-engine</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
```
Comment (trantienduchn) [inline server/task/pom.xml]: test scope?

Comment (mbaechler) [reply server/task/pom.xml]: good catch

**Pull request merged**